src/Bicep.Core/Diagnostics/DiagnosticBuilder.cs

```diff
@@ -1219,8 +1219,8 @@ public ErrorDiagnostic UnknownModuleReferenceScheme(string badScheme, ImmutableA
             public ErrorDiagnostic ExpectedProviderSpecification() => new(
                 TextSpan,
                 "BCP201",
-                "Expected a provider specification string of format \"<providerName>@<providerVersion>\" at this location.");
-
+                "Expected a provider identifier or a provider specification string of format \"br:<providerRegistryHost>/<providerRepositoryPath>@<providerVersion>\" or a string of format \"br/<providerAlias>:<providerName>@<providerVersion>\" at this location.");
+// Expected a provider identifier at this location.
```

Comment:
I wonder if it is worth discussing the possibility of phasing out the old format without causing breaking changes.

Reply:
It would be nice if we could figure how many users are using `dynamicTypeLoading`. If the affected user base is relatively small, proceeding with a breaking change could be justifiable.

Reply:
we agreed on doing a graceful phaseout, we have currently no way to measure usage of any experimental feature, including `dynamicTypeLoading`